src/generator.ts

```diff
@@ -249,6 +261,129 @@ const buildEndpoints = (config: Config, openapi: OpenAPIV3.Document): OpenAPIV3.
         responses: getResponses(operation, resourceSchemaPrefix),
       });
     });
+
+    // add relationship endpoints
+    _.forEach(resource.relationships, (relationship, relationshipName) => {
+      const path = `${getPath('getById', resource.plural)}/relationships/${relationshipName}`;
+      const relationshipSchemaPrefix = `${resourceSchemaPrefix}${getResourceSchemaPrefix(relationship.type)}Relationship`;
+      const isToMany = relationship.relationshipType === 'toMany';
+      const relationshipSchemaName = isToMany ? `${relationshipSchemaPrefix}Set` : relationshipSchemaPrefix;
+      const relationshipResponseSchemaName = `${relationshipSchemaName}Result`;
+
+      const relationshipSchema = getRelationshipSchema(relationship);
+      if (!_.has(openapi, `components.schemas.${relationshipSchemaPrefix}`)) {
+        _.set(openapi, `components.schemas.${relationshipSchemaPrefix}`, relationshipSchema);
+      }
+
+      const getResponseSchema = isToMany
+        ? getRelationshipSetResultSchema
+        : getRelationshipResultSchema;
+      const responseSchema = getResponseSchema(
+        resource,
+        relationshipName,
+        baseUrl,
+        relationshipSchemaPrefix,
+      );
+
+      _.set(openapi, `components.schemas.${relationshipResponseSchemaName}`, responseSchema);
+
+      _.set(openapi, `components.responses.${relationshipSchemaName}`, {
+        description: 'README',
+        content: {
+          'application/json': {
+            schema: {
+              $ref: `#/components/schemas/${relationshipResponseSchemaName}`,
+            },
+          },
+        },
+      });
+
+      const relationshipSchemaRef = {
+        $ref: `#/components/schemas/${relationshipSchemaPrefix}`,
+      };
+      const arrayData = {
+        type: 'array',
+        items: relationshipSchemaRef,
+      };
+      const singleData = {
+        type: 'object',
+        allOf: [
+          {
+            nullable: true,
+          },
+          relationshipSchemaRef,
+        ],
+      };
+      const relationshipRequestBody = {
+        content: {
+          'application/json': {
+            schema: {
+              type: 'object',
+              required: [
+                'data',
+              ],
+              properties: {
+                data: isToMany ? arrayData : singleData,
+              },
+            },
+          },
+        },
+      };
+
+      _.set(openapi, `components.requestBodies.${relationshipSchemaName}`, relationshipRequestBody);
+
+      const pathsObject = _.reduce(
+        ['get', 'patch', ...(isToMany ? ['post', 'delete'] : [])],
```

Comment:
I am not sure how the `...` operator is being used here. Usually I see it for variable length parameter definitions but that doesn't seem to be the case here. Can you describe what is happening here?

Reply:
From my understanding, when you use `function foo(...args) { doStuff(args); }`, it's called [rest parameters](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters) which like you said allows for a variable number of arguments. A call like `foo('a', 'b', 'c')` will compress the arguments into a list.

When you use `...object` or `...array`, it's called the [spread operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax) and it essentially does the opposite. It'll take an object or array and expand it. One common use case is for merging objects without using `Object.assign()`.
```js
obj1 = { a: 1, b: 2 };
obj2 = { a: 999, c: 3 };
obj3 = { ...obj1, ...obj2 } // { a: 999, b: 2, c: 3 } (obj2 comes after obj1 so it overwrites duplicate props in obj1
```
This can also be done with arrays.

Here, I'm using it to add in `'post'` and `'delete'` to the array only if `isToMany` is `true`. I think this syntax can be confusing though so I'll add a comment.